src/controller.coffee

```diff
@@ -958,6 +961,28 @@ define ['droplet-helper',
       # the hit test opportunity for this event.
       state.consumedHitTest = true
 
+  hook 'mousedown', 4, (point, event, state) ->
+    if state.consumedHitTest then return
+    if not @trackerPointIsInMain(point) then return
+
+    mainPoint = @trackerPointToMain point
+    hitTestResult = @hitTest mainPoint, @tree
+
+    if hitTestResult?
+      hitTestBlock = @view.getViewNodeFor hitTestResult
+      lineNumber = @findLineNumberAtCoordinate(mainPoint.y)
+
+      if hitTestBlock.addButtonRect? and hitTestBlock.addButtonRect.contains mainPoint
+        #console.log "Plus on: ", lineNumber
+        line = @mode.handleButton @getValue().split('\n')[lineNumber], 'add-button'
```

Comment:
handleButton should not work at the line-level, but at the block level.

You should probably pass handleButton three arguments
(1) the entire text for the block (which may be multiline)
(2) possibly the set of classes for the block (so that it's easy to tell which type of block it is)
(3) the command - 'add-button'

David